model/app.py

```diff
@@ -825,8 +886,41 @@ def remove_reactions(model, changes):
         model = model.copy()
         model.notes = deepcopy(model.notes)
         model = await modify_model(message, model)
-        db_key = await save_changes_to_db(model, wild_type_id, message)
-    return web.json_response(await respond(message, model, db_key))
+        mutated_model_id = await save_changes_to_db(model, wild_type_id, message)
+    return web.json_response(await respond(model, message, mutated_model_id))
+
+
+async def model_get_handler(request):
+    wild_type_id = request.match_info['model_id']
+    wild_type_model = find_in_memory(wild_type_id)
+    return web.json_response(model_to_dict(wild_type_model))
+
+
+async def model_diff_handler(request):
+    wild_type_id = request.match_info['model_id']
+    data = await request.json()
+
+    try:
+        message = data['message']
+    except KeyError:
+        return web.HTTPBadRequest()
+
+    wild_type_model = find_in_memory(wild_type_id)
+    if not wild_type_model:
+        return web.HTTPNotFound()
+    mutated_model_id = key_from_model_info(wild_type_id, message, version=1)
+    mutated_model = await restore_from_db(mutated_model_id)
+    logger.info(model_from_changes.cache_info())
+
+    if not mutated_model:
+        mutated_model = wild_type_model.copy()
+        # Remove this once model.copy gets fixed in Cameo
```

Comment:
Are we actually still using cameo? Should be cobrapy only really and in cobrapy `Model.copy()` *should* do the right thing.